// @astrojs/svelte/dist/index.js
import { svelte, vitePreprocess } from "@sveltejs/vite-plugin-svelte";
function getRenderer() {
  return {
    name: "@astrojs/svelte",
    clientEntrypoint: "@astrojs/svelte/client.js",
    serverEntrypoint: "@astrojs/svelte/server.js"
  };
}
function svelteIntegration(options) {
  return {
    name: "@astrojs/svelte",
    hooks: {
      "astro:config:setup": async ({ updateConfig, addRenderer }) => {
        addRenderer(getRenderer());
        updateConfig({
          vite: {
            plugins: [svelte(options), configEnvironmentPlugin()]
          }
        });
      }
    }
  };
}
function configEnvironmentPlugin() {
  return {
    name: "@astrojs/svelte:config-environment",
    configEnvironment(environmentName, options) {
      if (environmentName === "client" || (environmentName === "ssr" || environmentName === "prerender") && options.optimizeDeps?.noDiscovery === false) {
        return {
          optimizeDeps: {
            include: environmentName === "client" ? ["@astrojs/svelte/client.js"] : environmentName === "ssr" || environmentName === "prerender" ? ["svelte/server", "svelte/internal/server"] : [],
            exclude: environmentName === "ssr" || environmentName === "prerender" ? ["@astrojs/svelte/server.js"] : []
          }
        };
      }
    }
  };
}
export {
  svelteIntegration as default,
  getRenderer as getContainerRenderer,
  vitePreprocess
};
